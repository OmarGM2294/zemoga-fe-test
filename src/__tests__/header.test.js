import React from 'react'

import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import Header from '../components/header'
import Dropdown from '../components/dropdown'

describe('Header', () => {
  it('Should match snapshot', () => {
    const component = create(<Header />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Should show the search bar on click', () => {
    const component = shallow(<Header />)
    component.find('button[aria-label="Search"]').simulate('click')
    expect(component.find('input').length).toEqual(1)
  })

  it('Should not show the search bar on start', () => {
    const component = shallow(<Header />)

    expect(component.find('input').length).toEqual(0)
  })

  it('Should hide the search bar on click', () => {
    const component = shallow(<Header />)
    component.find('button[aria-label="Search"]').simulate('click')
    component.find('button[aria-label="Search"]').simulate('click')

    expect(component.find('input').length).toEqual(0)
  })

  it('Should have a dropdown component', () => {
    const component = shallow(<Header />)
    expect(component.find(Dropdown).length).toEqual(1)
  })
})
