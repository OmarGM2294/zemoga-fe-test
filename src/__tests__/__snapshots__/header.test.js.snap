// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `
<nav>
  <div>
    <h1>
      Rule of thumb.
    </h1>
    <div>
      <div
        onClick={[Function]}
      >
        <button>
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="Menu"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "visibility": "hidden",
                  "width": 0,
                }
              }
            />
          </div>
        </button>
      </div>
    </div>
    <ul
      className="undefined undefined"
    >
      <li>
        <a
          href="#"
        >
          Past Trials
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          How It Works
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Login / Sign Up
        </a>
      </li>
      <li>
        <form
          action="#"
        >
          <button
            aria-label="Search"
            onClick={[Function]}
            type="submit"
          >
            <div
              style={
                Object {
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "left": 0,
                  "margin": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <img
                alt="Search"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "width": 0,
                  }
                }
              />
            </div>
          </button>
        </form>
      </li>
    </ul>
  </div>
</nav>
`;
